mongodbatlas/data_source_mongodbatlas_project.go

```diff
@@ -140,7 +140,10 @@ func dataSourceMongoDBAtlasProjectRead(ctx context.Context, d *schema.ResourceDa
 
 	apiKeys, err := getProjectAPIKeys(ctx, conn, project.OrgID, project.ID)
 	if err != nil {
-		return diag.Errorf("error getting project's api keys (%s): %s", projectID, err)
+		var target *matlas.ErrorResponse
+		if errors.As(err, &target) && target.ErrorCode != "USER_UNAUTHORIZED" {
+			return diag.Errorf("error getting project's api keys (%s): %s", projectID, err)
+		}
 	}
```

Comment:
based on this change, `apiKeys` will still be empty when the request `getProjectAPIKeys` returns the error `USER_UNAUTHORIZED` so I would print a warning explaining why this field is empty.